app/stac_api/serializers.py

```diff
@@ -99,8 +99,9 @@ def filter_null(obj):
             for key, value in obj.items():
                 if isinstance(value, dict):
                     filtered_obj[key] = filter_null(value)
-                elif isinstance(value, list) and len(value) > 0:
-                    filtered_obj[key] = value
+                elif isinstance(value, list) and key != 'links':
```

Comment:
Maybe add a comment that links array might be empty at this point but that in the view auto generated links are added anyway.